geode-core/src/main/java/org/apache/geode/cache/query/internal/index/IndexManager.java

```diff
@@ -1392,6 +1392,19 @@ public String getCanonicalizedIteratorName(String definition) {
     return ((String) (this.canonicalizedIteratorNameMap.get(definition)));
   }
 
+  /**
+   * Update all indexes with maintenance status
+   *
+   */
+  public void updateIndexMaintenanceStatus(boolean ongoing) {
+    Collection<Index> allindexes = getIndexes();
+    for (Index index : allindexes) {
+      if (index.isMaintenanceOngoing() != ongoing) {
```

Comment:
We'd probably only want to set the maintenance status of the async indexes and not all indexes.
Also, the output from gfsh would be a snapshot in time.  It could say complete but a new put could have come in and it would be ongoing.  How useful is it to have this thing flicker based on put in a write heavy system?

Reply:
It is simpliest solution (and consistent), to update all indexes. Async indexes will have variable staus, and sync will allways be finished.